packages/aws-client/tests/test-S3.js

```diff
@@ -323,3 +330,58 @@ test('headObject() will retry if the requested key does not exist', async (t) =>
 
   await t.notThrowsAsync(promisedHeadObject);
 });
+
+const testEncodeTags = (t, input, expected) =>
+  t.is(encodeTags(input), expected);
+
+testEncodeTags.title = (_title, input, expected) => {
+  const inputJSON = JSON.stringify(input);
+  return `encodeTags(${inputJSON}) returns '${expected}'`;
+};
+
+test(testEncodeTags, [], '');
```

Comment:
I'm not sure I like this.   While it makes for less test boilerplate, and is pretty cool it also forces us to parse  testEncodeTags.title to know what's going on here if just reading the test source.

Reply:
I see that you and @mh81 don't care for this approach, so I will make the suggested changes.

However, I would like to consider this for discussion in an Arch Mtg, as I actually find this to be _more_ readable given that I can very easily scan the 3 related test cases to see the input and expected output, and then also very quickly and easily add more test cases, if necessary.

This is in line with every other testing framework I've used, where _parameterized_ tests are supported. The ava-specific syntax here may be slightly less clear than with other frameworks (perhaps particularly with non-JavaScript ones), but I still find it more clear and convenient, but again, that's perhaps for an Arch. discussion.